interp/notation.ml

```diff
@@ -1198,9 +1198,16 @@ let rec find_without_delimiters find (ntn_scope,ntn) = function
           find_without_delimiters find (ntn_scope,ntn) scopes
       end
   | LonelyNotationItem ntn' :: scopes ->
-      begin match ntn_scope, ntn with
-      | LastLonelyNotation, Some ntn when notation_eq ntn ntn' ->
-        Some (None, None)
+      begin match ntn with
```

Comment:
Can you add comments to explain what is going on here? This code seems tricky enough that it was buggy before this PR.

Reply:
Yes, this is rather tricky. I'll think at a documentation.

Reply:
Note also that this is overcautious: it does not detect if an only-notation is only-printing and if it is "safe" to use a scoped notation for the same string without a delimiter (but this is fixed in #12950). So, strictly speaking, in this PR, the fix could simply be:
```
        begin match ntn_scope with
        | LastLonelyNotation -> Some (None, None)
        | NotationInScope _ -> find_with_delimiters ntn_scope
        end
```
because the `find default_scope` is (if I'm not mistaken) always true.

Reply:
OK, tentative explanation made. Can you have a look?